packages/pilot/src/containers/RecipientTable/tableColumns.js

```diff
@@ -29,27 +38,24 @@ const columnData = data => (
   ))
 )
 
-const antecipationModel = (data) => {
-  if (data.automatic_anticipation_enabled === false) {
-    return 'manual'
-  }
-
-  if (data.automatic_anticipation_type === 'full') {
-    return 'automatic_volume'
-  }
-
-  if (data.automatic_anticipation_days === '10,25') {
-    return 'automatic_1025'
-  }
-
-  const antecipationDays = data.automatic_anticipation_days.split(',')
-
-  if (antecipationDays.length === 31) {
-    return 'automatic_dx'
-  }
+const isDX = allPass([
+  propIs(String, 'automatic_anticipation_days'),
+  pipe(
+    prop('automatic_anticipation_days'),
+    split(','),
+    length,
+    equals(31)
+  ),
+])
 
-  return 'custom'
-}
+const anticipationModel = cond([
+  [propEq('automatic_anticipation_type', 'compulsory'), always('compulsory')],
+  [propEq('automatic_anticipation_enabled', false), always('manual')],
+  [propEq('automatic_anticipation_type', 'full'), always('automatic_volume')],
+  [propEq('automatic_anticipation_days', '10,25'), always('automatic_1025')],
+  [isDX, always('automatic_dx')],
+  [T, always('custom')],
+])
```

Comment:
:clap: :clap: :clap: :clap: :clap:

Reply:
Boa @victorteodoro, poderias deixar estas condicionais do `cond` mais legiveis para os leigos como eu, ex: 
```suggestion
const whenAnticipationCompulsory = propEq('automatic_anticipation_type', 'compulsory')

const anticipationModel = cond([
	  [whenAnticipationCompulsory, always('compulsory')],
...
```
Aí só precisarias manter o padrão para as demais condições.
(PS o nome `whenAnticipationCompulsory` é só sugestão, deve ter algo melhor)

Reply:
@Deivis aproveitando, essa função deveria estar nesse arquivo? Se sim, posso mandar um teste na mesma página e adicionar o script `test` no `package.json`?

Reply:
@victorteodoro  o ideal seria mover esta chamada para o `cockpit` testar lá, mas como no caso desta página o retorno da chamada não está sendo transformado no `cockpit` acaba gerando esta tratativa. Por hora podes deixar como está (apenas para dar continuidade no PR) mas sugiro fortemente que movam estas tratativas para dentro do `cockpit` para que estas funções de componentes e containers fiquem mais simples 😄

Reply:
Show! Era o que achava mesmo então! Valeu cara :+1: